Sources/MongoSwift/BSON/Document+Sequence.swift

```diff
@@ -241,21 +234,25 @@ public class DocumentIterator: IteratorProtocol {
 
     /// Returns the values from the iterator's current position to the end. The iterator
     /// will be exhausted after this property is accessed.
-    internal var values: [BSONValue?] {
-        var values = [BSONValue?]()
+    internal var values: [BSONValue] {
+        var values = [BSONValue]()
         while self.advance() { values.append(self.currentValue) }
         return values
     }
 
     /// Returns the current value (equivalent to the `currentValue` property) or throws on error.
-    internal func safeCurrentValue() throws -> BSONValue? {
+    internal func safeCurrentValue() throws -> BSONValue {
         switch self.currentType {
         case .symbol:
             return try Symbol.asString(from: self)
         case .dbPointer:
             return try DBPointer.asDocument(from: self)
         default:
-            return try DocumentIterator.bsonTypeMap[currentType]?.init(from: self)
+            if let curVal = try DocumentIterator.bsonTypeMap[currentType]?.from(iterator: self) {
```

Comment:
similar to my other comment, prefer `guard let...` and then return when the `else` throws.